### ISSUE TITLE
First sprint results

### ISSUE THREAD

**Issue by KirillDogadin-std** [KirillDogadin-std/Lab-Semantic#7]
Lets summarize the advancements made by now and those that will be made by Fri.

aka presentation content. 

I suggest using a MD based slides app instead of full power-point mode. 

Edit the comment here so that i can c-paste it and here we go: a presentation's ready. 

---

You may also merge demos if you want to prepare them. PR is #8

**Comment by KirillDogadin-std** [comment]
## Report №1
---
### **Preparation phase**

- fortify knowledge in `pytorch`
- study the codebase in more detail
	- understand the `tensorflow` code

---
### **Implementation phase**
- begin implementation
- start on 
	- `Multi_KE` model code refactoring
	- `entity embedding` code refactoring
---
### **Tools used**

- github for code storage & more effective communication.
- Jupyter to test out code snippets.

---

### **Coming soon™** 

- Polishing & testing of `model`

**Comment by 4rash-uni** [comment]
it is fine.

**Comment by KirillDogadin-std** [comment]
### **Preparation phase**

- fortify knowledge in `pytorch`
- study the codebase in more detail
	- understand the `tensorflow` code

---
### **Implementation phase**
- start on refactoring:
	- `Multi_KE` 
	- `losses`
	- `data model`
	- `data loader`
	- `literal encoder`
	- `knowledge graphs`
---
### **Challenges**
- PyTorch vs TensorFlow 1.x
	- dynamic vs static computational graph definition
	- different parameter convention and default value e.g. `batchnorm`
	- dimensions ordering i.e. channel first vs channel last
	- no corresponding equivalent for some functions
---
### **Challenges**
- large dataset and high prerocessing cost
	- up to 32GB wo training
